src/tests/lvm/lvm_ha_controller/lvm_ha_controller_test.go

```diff
@@ -0,0 +1,274 @@
+		When Lvm controller is scaled down to zero replica
+		And Lvm volume(pvc) is deployed
+		Then Lvm volume(pvc) should remain in pending state
+		When Lvm controller is scaled to two replica via helm upgrade
+		Then lvm volume(pvc) should transition to bound state
+		And Deploy fio application to use lvm volume
+		And Taint all worker node with NoSchedule taint
+		When Delete one of the lvm controller pod
+		Then Lvm node lease should point to running lvm controller pod
+*/
+
+var nodeConfig lvm.LvmNodesDevicePvVgConfig
+var defFioCompletionTime = 120 // in seconds
+var nodesWithoutTaint []string
+var lvmControllerOrgReplica int32
+var volumeProvisionErrorMsg = "waiting for a volume to be created"
+
+func controllerHaTest(decor string, engine common.OpenEbsEngine, volType common.VolumeType, fstype common.FileSystemType, volBindModeWait bool) {
+
+	e2e_config := e2e_config.GetConfig().Product
+	app := k8stest.FioApplication{
+		Decor:                          decor,
+		VolSizeMb:                      4096,
+		OpenEbsEngine:                  engine,
+		VolType:                        volType,
+		FsType:                         fstype,
+		Loops:                          5,
+		VolWaitForFirstConsumer:        volBindModeWait,
+		SkipPvcVerificationAfterCreate: true,
+	}
+
+	loopDevice := e2e_agent.LoopDevice{
+		Size:   10737418240,
+		ImgDir: "/tmp",
+	}
+
+	workerNodes, err := lvm.ListLvmNode(common.NSOpenEBS())
+	Expect(err).ToNot(HaveOccurred(), "failed to list worker node")
+
+	nodeConfig = lvm.LvmNodesDevicePvVgConfig{
+		VgName:        "lvmvg",
+		NodeDeviceMap: make(map[string]e2e_agent.LoopDevice), // Properly initialize the map
+	}
+	for _, node := range workerNodes {
+		nodeConfig.NodeDeviceMap[node] = loopDevice
+	}
+
+	logf.Log.Info("setup node with loop device, pv and vg", "node config", nodeConfig)
+	err = nodeConfig.ConfigureLvmNodesWithDeviceAndVg()
+	Expect(err).ToNot(HaveOccurred(), "failed to setup node")
+
+	// setup sc parameters
+	app.Lvm = k8stest.LvmOptions{
+		VolGroup:      nodeConfig.VgName,
+		Storage:       "lvm",
+		ThinProvision: common.No,
+	}
+
+	// list nodes without NoSchedule taint
+	nodesWithoutTaint, err = k8stest.ListNodesWithoutNoScheduleTaint()
+	Expect(err).ToNot(HaveOccurred(), "failed to list nodes without NoSchedule taint")
+	logf.Log.Info("nodes without NoSchedule taint", "nodes", nodesWithoutTaint)
+
+	// get the no of replicas in lvm-controller deployment
+	// Scale down the lvm-controller deployment
+	// Check that lvm-controller pods has been terminated successfully
+	logf.Log.Info("Scale down and get the no of replicas in lvm-controller deployment")
+	lvmControllerName := e2e_config.LvmEngineControllerDeploymentName
+	lvmControllerOrgReplica, err = k8sinstall.ScaleLvmControllerViaHelm(0)
+	Expect(err).To(BeNil(), "failed to scale down deployment %s, error: %v", lvmControllerName, err)
+	Expect(lvmControllerOrgReplica).ShouldNot(BeZero(), "lvm controller replica count should not be zero")
```

Comment:
if called with scale with 0 replica, then why it should not be zero here ?

Reply:
It's because if you check `ScaleLvmControllerViaHelm` function , it return actual original replica count before scale lvm controller operation